src/overlays/actors/ovl_Demo_Tre_Lgt/z_demo_tre_lgt.c

```diff
@@ -28,24 +47,123 @@ const ActorInit Demo_Tre_Lgt_InitVars = {
     (ActorFunc)DemoTreLgt_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_06007D78;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Tre_Lgt/DemoTreLgt_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Tre_Lgt/DemoTreLgt_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Tre_Lgt/func_808E0EE8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Tre_Lgt/func_808E0EF4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Tre_Lgt/func_808E0F4C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Tre_Lgt/func_808E0FE0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Tre_Lgt/DemoTreLgt_Update.s")
+static TransformUpdateIndex* sBoxLightAnimations[] = {
+    &gBoxLight0CurveAnim,
+    &gBoxLight1CurveAnim,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Tre_Lgt/func_808E12A4.s")
+static DemoTreLgtActionFunc sActionFuncs[] = {
+    DemoTreLgt_Wait,
+    DemoTreLgt_Animate,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Demo_Tre_Lgt/DemoTreLgt_Draw.s")
+void DemoTreLgt_Init(Actor* thisx, GlobalContext* globalCtx) {
+    DemoTreLgt* this = (DemoTreLgt*)thisx;
+    SkelCurve_Init(globalCtx, &this->skelCurve, &gBoxLightCurveSkel, sBoxLightAnimations[0]);
+    this->colorAlpha1 = 255;
+    this->colorAlpha2 = 255;
+    this->status = 0;
+    if (gSaveContext.playerForm == 0) {
```

Comment:
Ah, you helped me figure out what's different about the light animation.

So in Ocarina of Time, there are two big chest opening animations. The key difference between them is that after Child Link first cracks open the chest, he's too short to push it any further up, so he closes the lid again before giving that last push that opens the chest up, whereas Adult Link is taller, so he opens it up wider on the first crack and and gives a quick push to fully open the chest.

Thus you have different light animations; the adult link one rises high as soon as Link cracks the chest open, whereas the child one stays lower for longer to prevent clipping when the lid shuts back closed. 

The weird part about this code is that Zora Link uses the adult link animation, so the chest light is technically "bugged" for Zora Link.